src/bcf/record.rs

```diff
@@ -74,6 +74,45 @@ impl NumericUtils for i32 {
 
 /// A BCF record.
 /// New records can be created by the `empty_record` methods of `bcf::Reader` and `bcf::Writer`.
+/// # Example
+///   - Obtaining 0-based locus index of the VCF record.
+///   - Obtaining alleles of the VCF record.
+///   - calculate alt-allele dosage in a mutli-sample VCF / BCF
+///
+/// ```
+/// use crate::rust_htslib::bcf::{Reader, Read};
+/// use std::convert::TryFrom;
+///
+/// let path = &"test/test_string.vcf";
+/// let mut bcf = Reader::from_path(path).expect("Error opening file.");
+///
+/// for (i, record_result) in bcf.records().enumerate() {
+///     let mut record = record_result.expect("Fail to read record");
+///     let mut s = String::new();
+///      for allele in record.alleles() {
+///          for c in allele {
+///              s.push(char::from(*c))
+///          }
+///          s.push(' ')
+///      }
+///     println!("Locus: {}, Alleles: {}", record.pos(), s);  // 0-based position, Alleles
+///     let sample_count = usize::try_from(record.sample_count()).unwrap();
+///     let mut n_ref = vec![0; sample_count];
+///     let mut n_alt = vec![0; sample_count];
+///     let gts = record.genotypes().expect("Error reading genotypes");
+///     for sample_index in 0..sample_count {
+///         // for each sample
+///         for gta in gts.get(sample_index).iter() {
+///             // for each allele
+///             match gta.index() {
+///                 Some(0) => n_ref[sample_index] += 1, // reference allele
+///                 Some(_) => n_alt[sample_index] += 1, // alt allele
+///                 None => break,                       // missing allele
+///             }
+///         }
+///     }
+/// }
+/// ```
```

Comment:
I think, this best goes into the main file of the module (`src/bcf/mod.rs`) as module-level documentation (`//!`) that will eventually show up in the docs right here: https://docs.rs/rust-htslib/0.32.0/rust_htslib/bcf/index.html